Tests/Functional/SelectedTest/main.swift

```diff
@@ -0,0 +1,58 @@
+// RUN: %{swiftc} %s -o %{built_tests_dir}/SelectedTest
+// RUN: %{built_tests_dir}/SelectedTest SelectedTest.ExecutedTestCase/test_foo > %t || true
+// RUN: %{built_tests_dir}/SelectedTest SelectedTest.ExecutedTestCase >> %t || true
+// RUN: %{built_tests_dir}/SelectedTest >> %t || true
+// RUN: %{xctest_checker} %t %s
+//// Output from the first invocation (running a single test method)
```

Comment:
You should be able to handle these cases as `llvm-lit` would: by using a custom prefix. That is, run the checker once for each variation:

- `%{xctest_checker} --prefix "// CHECK-METHOD:" %T/one_test_method`
- `%{xctest_checker} --prefix "// CHECK-TESTCASE:" %T/one_test_case`
- `%{xctest_checker} --prefix "// CHECK-ALL:" %T/all`

(Here I'm assuming you've piped the output into files named `one_test_method`, etc., in a temporary directory `%T`. If our `lit.cfg` doesn't provide a temporary directory substitution we should add one!)